### DISCUSSION TITLE
Set allure-results directory at runtime

### DISCUSSION THREAD

**Discussion by magarich228** [allure-framework/allure2#2536] [Questions & Support] [answered]
Hello. Saw discussion #2154, completed in 2023. Is there no way to set the results directory at runtime now? 

In my case, my tests can be executed simultaneously in multiple threads. I am trying to come up with a solution that ensures that the results of one test run are placed in a directory with a unique name.. But studying the sources and features of Allure I am a bit stumped.

**Comment by baev** [comment]
Why do you need to have different result directories?

**Reply by magarich228** [reply]
That's the way my customer wants it. He needs to be able to generate a report for each individual test run.

**Reply by baev** [reply]
Could you please provide a bit more context? What do you mean by saying "a report for each individual test run"?

We don't want to allow changing the results directory in runtime because it creates many additional complications, e.g., the real-time reporting feature from Allure TestOps or Allure 3 (WIP) will not work properly.

**Reply by magarich228** [reply]
Unfortunately I can't attach code samples. But the context of the phrase "a report for each individual test run" is that my NUnit tests are run from code, they can run many times. And after each run I need to separate the files from the allure-results directory into a separate directory related to that run.

**Reply by baev** [reply]
Are you running NUnit in separate processes? In that case, you can set the `ALLURE_CONFIG` environment variable pointing to the config file location and create a separate config file (e.g. in the temp directory) per run.

**Reply by magarich228** [reply]
Unfortunately no, I'm limited to threads only.

**Reply by baev** [reply]
Can you provide a clue on how you initiate NUnit and run the tests?

**Reply by magarich228** [reply]
Yes, nothing out of the ordinary, I use NUnitTestAssemblyRunner. I load the necessary assemblies and run all or some tests from them.

**Reply by baev** [reply]
please follow https://github.com/allure-framework/allure-csharp/issues/497

**Comment by baev** [comment, marked as answer]
It's not yet possible to set config programmatically. The issue to follow — https://github.com/allure-framework/allure-csharp/issues/497

Suggested workaround:

Run NUnit in separate processes, not threads and provide different config values via `ALLURE_CONFIG` environment variable.
